tools/bin/hawqregister

```diff
@@ -126,182 +127,319 @@ def option_parser_yml(yml_file):
+                sys.exit(1)
+            check_policy_consistency()
+            check_bucket_number()
+            existed_files, existed_sizes = self._get_files_in_hdfs(self.filepath)
+            existed_info = {}
+            for k, fn in enumerate(existed_files):
+                existed_info[fn] = existed_sizes[k]
+            for k, fn in enumerate(self.files):
+                if fn not in existed_files:
+                    logger.error('Can not register --repair since giving non-existed files.')
+                    sys.exit(1)
+                if self.sizes[k] > existed_info[fn]:
+                    logger.error('Can not register --repair since giving larger file size.')
+                    sys.exit(1)
 
-def get_files_in_hdfs(filepath):
-    '''Get all the files refered by 'filepath', which could be a file or a directory containing all the files'''
-    files = []
-    sizes = []
-    hdfscmd = "hdfs dfs -test -e %s" % filepath
-    result = local_ssh(hdfscmd, logger)
-    if result != 0:
-        logger.error("Path '%s' does not exist in hdfs" % filepath)
-        sys.exit(1)
-    hdfscmd = "hdfs dfs -ls -R %s" % filepath
-    result, out, err = local_ssh_output(hdfscmd)
-    outlines = out.splitlines()
-    # recursively search all the files under path 'filepath'
-    for line in outlines:
-        lineargs = line.split()
-        if len(lineargs) == 8 and lineargs[0].find ("d") == -1:
-            files.append(lineargs[7])
-            sizes.append(int(lineargs[4]))
-    if len(files) == 0:
-        logger.error("Dir '%s' is empty" % filepath)
-        sys.exit(1)
-    return files, sizes
-
-
-def check_parquet_format(files):
-    '''Check whether the file to be registered is parquet format'''
-    for f in files:
-        hdfscmd = 'hdfs dfs -du -h %s | head -c 1' % f
-        rc, out, err = local_ssh_output(hdfscmd)
-        if out == '0':
-            continue
-        hdfscmd = 'hdfs dfs -cat %s | head -c 4 | grep PAR1' % f
-        result1 = local_ssh(hdfscmd, logger)
-        hdfscmd = 'hdfs dfs -cat %s | tail -c 4 | grep PAR1' % f
-        result2 = local_ssh(hdfscmd, logger)
-        if result1 or result2:
-            logger.error('File %s is not parquet format' % f)
+        if self.mode == 'second_exist':
+            if self.tabledir.strip('/') == self.filepath.strip('/'):
+                logger.error('Files to be registeted in this case should not be the same with table path.')
+                sys.exit(1)
+
+        self.do_not_move, self.files_update, self.sizes_update = False, [], []
+        if self.mode == 'force':
+            existed_files, _ = self._get_files_in_hdfs(self.tabledir)
+            if len(self.files) == len(existed_files):
+                if sorted(self.files) != sorted(existed_files):
+                    logger.error('In this case, you should include previous table files.\nOtherwise you should drop the previous table before registering --force.')
```

Comment:
In force mode, you should include existing table files in .yml configuration file. Otherwise you should drop ...